code/datums/controllers/process/machines.dm

```diff
@@ -68,14 +69,17 @@ datum/controller/process/machines
 
 		for (var/i in 1 to PROCESSING_MAX_IN_USE)
 			var/list/machlist = src.machines[i]
+
 			for(var/X in machlist[(src.ticker % (1<<(i-1)))+1])
 				if(!X) continue
 				var/obj/machinery/machine = X
 				if( machine.z == 4 && !Z4_ACTIVE ) continue
 		#ifdef MACHINE_PROCESSING_DEBUG
 				var/t = world.time
 		#endif
-				machine.process()
+				mult = clamp(TIME - machine.last_process, machine.base_tick_spacing*(2**(machine.processing_tier-1)), machine.cap_base_tick_spacing*(2**(machine.processing_tier-1))) / (machine.base_tick_spacing*(2**(machine.processing_tier-1)))
```

Comment:
It'd be much more readable if you put the target machine loop time into a separate var first as it repeats three times in the expression

Reply:
Okay I moved the base and cap vars to vars in the loop. Should they be defined every loop like this, or defined at the top and reassigned every loop?